api/src/main/java/ca/bc/gov/educ/api/batchgraduation/controller/JobLauncherController.java

```diff
@@ -641,4 +644,86 @@ private PsiDistributionSummaryDTO validateInputPsiDisRun(PsiCredentialRequest ps
+    @PreAuthorize(PermissionsConstants.RUN_GRAD_ALGORITHM)
+    @Operation(summary = "Run Monthly Distribution Runs", description = "Run Monthly Distribution Runs", tags = { "Distribution" })
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"),@ApiResponse(responseCode = "500", description = "Internal Server Error")})
+    public ResponseEntity<Void> notifyDistributionJobIsCompleted(
+            @RequestParam(name = "batchId", defaultValue = "0") Long batchId,
+            @RequestParam(name = "status", defaultValue = "success") String status) {
+
+        BatchGradAlgorithmJobHistoryEntity jobHistory = gradBatchHistoryService.getGradAlgorithmJobHistory(batchId);
+        String jobType = jobHistory.getJobType();
+
+        if (StringUtils.equalsIgnoreCase(status, "success")) {
+            List<StudentCredentialDistribution> cList = gradBatchHistoryService.getStudentCredentialDistributions(batchId);
+
+            // update graduation_student_record & student_certificate
+            Map<String, ServiceException> unprocessed = updateBackStudentRecords(cList, batchId, getActivitCode(jobType));
+            if (!unprocessed.isEmpty()) {
+                status = BatchStatusEnum.FAILED.name();
+                this.handleUnprocessedErrors(unprocessed);
+            } else {
+                status = BatchStatusEnum.COMPLETED.name();
+            }
+        } else {
+            status = BatchStatusEnum.FAILED.name();
+        }
+
+        // update status for batch job history
+        Date endTime = new Date(System.currentTimeMillis());
+        jobHistory.setEndTime(endTime);
+        jobHistory.setStatus(status);
+        jobHistory.setJobParameters(populateJobParametersDTO(jobType, null));
+        gradBatchHistoryService.saveGradAlgorithmJobHistory(jobHistory);
+
+        return ResponseEntity.ok(null);
+    }
+
+    private Map<String, ServiceException> updateBackStudentRecords(List<StudentCredentialDistribution> cList, Long batchId, String activityCode) {
+        Map<String, ServiceException> unprocessedStudents = new HashMap<>();
+        cList.forEach(scd-> {
+            try {
+                final String token = restUtils.getTokenResponseObject().getAccess_token();
+                restUtils.updateStudentCredentialRecord(scd.getStudentID(),scd.getCredentialTypeCode(),scd.getPaperType(),scd.getDocumentStatusCode(),activityCode,token);
+                restUtils.updateStudentGradRecord(scd.getStudentID(),batchId,activityCode,token);
+            } catch (Exception e) {
+                unprocessedStudents.put(scd.getStudentID().toString(), (ServiceException) e);
+            }
+        });
+        return unprocessedStudents;
+    }
+
+    private String getActivitCode(String jobType) {
+        String activityCode = "MONTHLYDIST";
+        if(StringUtils.isNotBlank(jobType)) {
+            switch (jobType) {
+                case "DISTRUN" -> activityCode = "MONTHLYDIST";
```

Comment:
Hi Jinil, please use already defined DISTRUN, DISTRUN_YE, DISTRUN_SUPP and NONGRADRUN constants from this class.

Reply:
Done.  I committed the code.